packages/davinci-client/src/lib/davinci.utils.ts

```diff
@@ -1,11 +1,20 @@
+/**
+ * Import the used types
+ */
+import { SingleValueCollector } from './collector.types';
 import { DaVinciRequest } from './davinci.types';
-import { NodeState } from './node.types';
+import { NextNode } from './node.types';
 
-export function transformSubmitRequest(node: NodeState): DaVinciRequest {
+/**
+ * @function transformSubmitRequest - Transforms a NextNode into a DaVinciRequest for form submissions
+ * @param {NextNode} node - The node to transform into a DaVinciRequest
+ * @returns {DaVinciRequest} - The transformed request object
+ */
+export function transformSubmitRequest(node: NextNode): DaVinciRequest {
   // Filter out ActionCollectors as they are not used in form submissions
-  const collectors = node.client.collectors?.filter(
+  const collectors = node.client?.collectors?.filter(
     (collector) => collector.category === 'SingleValueCollector',
-  );
+  ) as SingleValueCollector[];
```

Comment:
I think in TS 5.5 or > we shouldn't need this, is the collectors typed appropriately within the node?